lua/entities/terminator_nextbot_zambieglass/shared.lua

```diff
@@ -98,25 +115,78 @@ function ENT:Think()
 
 end
 
+// Default shard count
+ENT.iShards = 10
+
+local sound_Add = sound.Add
+sound_Add {
+	name = "nextbotZambies_GlassBreakA",
+	level = 85,
+	sound = {
+		"physics/glass/glass_largesheet_break1.wav",
+		"physics/glass/glass_largesheet_break2.wav",
+		"physics/glass/glass_largesheet_break3.wav"
+	}
+}
+sound_Add {
+	name = "nextbotZambies_GlassBreakB",
+	level = 80,
+	pitch = 110,
+	sound = {
+		"physics/glass/glass_sheet_break1.wav",
+		"physics/glass/glass_sheet_break2.wav",
+		"physics/glass/glass_sheet_break3.wav"
+	}
+}
+
+local SHARD_MODELS = {
+	"models/gibs/glass_shard01.mdl",
+	"models/gibs/glass_shard02.mdl",
+	"models/gibs/glass_shard03.mdl",
+	"models/gibs/glass_shard04.mdl",
+	"models/gibs/glass_shard05.mdl",
+	"models/gibs/glass_shard06.mdl",
+}
+local SHARD_MODELS_LENGTH = 6
+
+local string_lower = string.lower
+local tonumber = tonumber
+
+function ENT:KeyValue( sKey, sValue )
+	if string_lower( sKey ) == "ishards" then
+		local f = tonumber( sValue )
+		if f then self.iShards = f end
+		return
+	end
+	return BaseClass.KeyValue( self, sKey, sValue )
+end
+
+local EffectData = EffectData
+local util_Effect = util.Effect
+local ents_Create = ents.Create
+local math = math
+local math_random = math.random
+local VectorRand = VectorRand
+local AngleRand = AngleRand
+local COLLISION_GROUP_DEBRIS = COLLISION_GROUP_DEBRIS
+local math_Rand = math.Rand
+local SafeRemoveEntityDelayed = SafeRemoveEntityDelayed
+
 function ENT:GlassZambDie()
     local pos = self:WorldSpaceCenter()
 
-    self:EmitSound( "physics/glass/glass_largesheet_break" .. math.random( 1, 3 ) .. ".wav", 85, 100 )
-    self:EmitSound( "physics/glass/glass_sheet_break" .. math.random( 1, 3 ) .. ".wav", 80, 110 )
-
-    local glassGibs = {
-        "models/gibs/glass_shard01.mdl",
-        "models/gibs/glass_shard02.mdl",
-        "models/gibs/glass_shard03.mdl",
-        "models/gibs/glass_shard04.mdl",
-        "models/gibs/glass_shard05.mdl",
-        "models/gibs/glass_shard06.mdl",
-    }
+    self:EmitSound "nextbotZambies_GlassBreakA"
+    self:EmitSound "nextbotZambies_GlassBreakB"
 
-    for _ = 1, 10 do
-        local gib = ents.Create( "prop_physics" )
+	local iShards = self.iShards
+	if iShards <= 0 then return end

```

Comment:
When will this be 0?
Will some code ever spawn a glass zombie, who's gimmick is glass shards, with 0 glass shards?
Probably never, should cross that bridge when we get there.